hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -75,6 +75,9 @@ abstract class HoodieBaseHadoopFsRelationFactory(val sqlContext: SQLContext,
   protected lazy val basePath: StoragePath = metaClient.getBasePath
   protected lazy val partitionColumns: Array[String] = tableConfig.getPartitionFields.orElse(Array.empty)
 
+  //placeholder until pr to persist field types lands
+  protected lazy val partitionColumnsToRead: Array[String] = if (shouldExtractPartitionValuesFromPartitionPath) Array.empty else tableConfig.getPartitionFields.orElse(Array.empty)
+
```

Comment:
@lokeshj1703  replace `tableConfig.getPartitionFields.orElse(Array.empty)` with partition cols that are timestamp keygen type

Reply:
Could we add a logic in this PR that only used `tableConfig.getPartitionFields` if timestamp or custom key generator is used, since other key generators do not change the partition value so it does not matter whether the partition values come from the file path or the data file?

Reply:
In @lokeshj1703 's PR, we can further restrain the condition if the simple-typed partition is used in custom key generator.

Reply:
addressed restricting to only custom and timestamp keygen.